source/common/network/tcp_listener_impl.cc

```diff
@@ -59,7 +59,7 @@ void TcpListenerImpl::onSocketEvent(short flags) {
       break;
     }
 
-    if (rejectCxOverGlobalLimit()) {
+    if (rejectCxOverGlobalLimit() || reject_fraction_.isRandomizedActive(random_)) {
```

Comment:
Once https://github.com/envoyproxy/envoy/pull/13305 lands (which I think will before this) I thin it'd be useful to tag why the connection was closed.

Reply:
I don't think we can tag anything here since this happens before an Envoy connection object is created, so there's nothing to annotate.